src/Tasks/Microsoft.Common.CurrentVersion.targets

```diff
@@ -119,30 +127,35 @@ Copyright (C) Microsoft Corporation. All rights reserved.
 
     IntermediateOutputPath:
     This is the full intermediate Output Path, and is derived from BaseIntermediateOutputPath, if none specified
-    (eg. obj\debug). If this property is overridden, then setting BaseIntermediateOutputPath has no effect.
+    (eg. obj\Debug). If this property is overridden, then setting BaseIntermediateOutputPath has no effect.
+
+    Ensure any and all path property has a trailing slash, so it can be concatenated.
     -->
 
   <PropertyGroup>
-    <!-- Ensure any OutputPath has a trailing slash, so it can be concatenated -->
-    <OutputPath Condition="'$(OutputPath)' != '' and !HasTrailingSlash('$(OutputPath)')">$(OutputPath)\</OutputPath>
+    <BaseOutputPath Condition=" '$(BaseOutputPath)' == '' ">bin\</BaseOutputPath>
+    <BaseOutputPath Condition="!HasTrailingSlash('$(BaseOutputPath)')">$(BaseOutputPath)\</BaseOutputPath>
```

Comment:
```suggestion
    <BaseOutputPath>$([MSBuild]::EnsureTrailingSlash('$([MSBuild]::ValueOrDefault('$(BaseOutputPath)', 'bin\'))'))</BaseOutputPath>
```

?

Reply:
The lines are similar to how it was presented in the SDK targets and to `BaseIntermediateOutputPath` lines.

I have the refactored version of the targets (in a separate branch). If it's okay with you, I can create a separate PR for those changes too!

Reply:
Sure, you can follow the convention here.

Reply:
OK. I'll also group the output logic together.

Reply:
I think I'd prefer inline changes without moves to make the changes super clear.